fetch_mail.py

```diff
@@ -0,0 +1,160 @@
+"""
+Method1
+-------
+    1. imaplib module
+        -> Establish a connection
+        -> Select the mailbox
+        -> Search for unseen mails
+        -> fetch their id
+        -> get raw data of each id
+        -> perform regex to find msg. patterns
+        -> save into a file
+        -> quit()
+"""
+
+import imaplib  # For reading messages
+import logging  # For catching logs
+import email
+import re
+import email.policy
+import socket  # Require for imaplib. connection
+
+
+class LPForum:
+    def __init__(self, email, password):
+        self.email_user = email
+        self.email_pass = password
+
+    def __str__(self):
+        return str(self.email_user)
+
+    def login(self):
+        """ Creating a secure connection over SSL socket
+        Return an imap4 object to "mail" variable """
+        try:
+            mail = imaplib.IMAP4_SSL("imap.gmail.com", 993)
+
+            # login to user account
+            mail.login(self.email_user, self.email_pass)
+            return mail
+        except imaplib.IMAP4.error as err:
+            logging.info(err)
+            return False
+
+    def get_message_id(self, obj):
+        """ Method for Return the number of mails
+        specified in the mailbox. """
+
+        obj.select(mailbox="INBOX")  # Select a mailbox first
+        _, msgnums = obj.search(
+            "utf-8", "(Unseen)", "FROM", "linkinpark@discoursemail.com"
+        )  # Search message inside mailbox
+        logging.info(f"Found unseen mail list : {msgnums}")
+        if len(msgnums[0]) > 0:
+            return msgnums
+        return False
+
+    def fetch_raw_message(self, obj, message_id):
+        """ Extracting raw message from message id's.
+        message_id contains id's of email messages
+        by linkinpark@discoursemail. 
+        (RFC822) is a msg. format, defines => header, payload
+        """
+
+        ids = message_id[0].split()
+        data = []
+        for id_ in ids:
+            typ, response = obj.fetch(id_, "(RFC822)")
+            data.append(response[0][1])
+
+        if typ == "OK":
```

Comment:
This is only checking the last result. If you had many ids, and some of them weren't OK, this check would be missed for them.

Instead, you can define `errors = []`, and check in the loop. Then if that errors.length != 0 handle that case.

Maybe this is alright, not really sure whats happening here.

Reply:
I guess this is alright because each id represents the email (in my case, these id's represents the unseen email came from linkinpark@discoursemail.com. 
Now, imaplib.fetch() helps in fetching the very raw data of each id and returns a tuple that contains a status code OK (if succeed) and a list in which the id with RFC and email message data are stored. 

Check -> ('OK', [(b'34395 (RFC822 {37833}', b'Delivered-To: h3c.initiativ...........])
This raw data is very hard to read and that's why I'm using another module called email